interface/modules/zend_modules/module/Carecoordination/src/Carecoordination/Model/EncounterccdadispatchTable.php

```diff
@@ -2192,62 +2197,88 @@ public function getCCDAComponents($type)
     {
         $content = base64_decode($content);
-        $file_path = '';
-        $docid = '';
-        $revid = '';
-        if ($GLOBALS['document_storage_method'] == 1) {
-            $couch = new CouchDB();
-            $docid = $couch->createDocId('ccda');
-            $binaryUuid = UuidRegistry::uuidToBytes($docid);
-            if ($GLOBALS['couchdb_encryption']) {
-                $encrypted = 1;
-                $cryptoGen = new CryptoGen();
-                $resp = $couch->save_doc(['_id' => $docid, 'data' => $cryptoGen->encryptStandard($content, null, 'database')]);
-            } else {
-                $encrypted = 0;
-                $resp = $couch->save_doc(['_id' => $docid, 'data' => base64_encode($content)]);
-            }
-            $docid = $resp->id;
-            $revid = $resp->rev;
-        } else {
-            $binaryUuid = (new UuidRegistry(['table_name' => 'ccda']))->createUuid();
-            $file_name = UuidRegistry::uuidToString($binaryUuid);
-            $file_path = $GLOBALS['OE_SITE_DIR'] . '/documents/' . $pid . '/CCDA';
-            if (!is_dir($file_path)) {
-                if (!mkdir($file_path, 0777, true) && !is_dir($file_path)) {
-                    // php Exception extends RunTimeException
-                    throw new Exception(sprintf('Directory "%s" was not created', $file_path));
-                }
-            }
+        $document = new \Document();
+
+        // we need to populate the category id based upon the document_type
+        // TOC -> CCDA folder
+        // CCD -> TOC
+        //
+        $categoryId = QueryUtils::fetchSingleValue(
+            'Select `id` FROM categories WHERE name=?',
+            'id',
+            [self::CCDA_DOCUMENT_FOLDER]
+        );
 
-            $fccda = fopen($file_path . "/" . $file_name, "w");
-            if ($GLOBALS['drive_encryption']) {
-                $encrypted = 1;
-                $cryptoGen = new CryptoGen();
-                fwrite($fccda, $cryptoGen->encryptStandard($content, null, 'database'));
-            } else {
-                $encrypted = 0;
-                fwrite($fccda, $content);
+        if ($categoryId === false) {
+            throw new RuntimeException("document category id does not exist in system");
+        }
+
+        $binaryUuid = (new UuidRegistry(['table_name' => 'ccda']))->createUuid();
+        $file_name = UuidRegistry::uuidToString($binaryUuid);
+        $mimeType = "text/xml";
+
+        $higherLevelPath = "";
+        $pathDepth = 1;
+        $owner = $_SESSION['user_id'];  // userID of who is creating the document...
+        $tmpFile = null;
+        $expirationDate = null;
+
+        try {
+            \sqlBeginTrans();
+
+            // set the foreign key so we can track documents connected to a specific export
+            $result = $document->createDocument(
+                $pid,
+                $categoryId,
+                $file_name,
+                $mimeType,
+                $content,
```

Comment:
To simplify code, can stop after $content with the parameters. All the rest will work via default ($owner will be set to $_SESSION['authUserID'] within the function).